### ISSUE TITLE
[SUGGESTION] esproj support for csdevkit

### ISSUE THREAD

**Issue by jpringle1** [microsoft/vscode-dotnettools#157]
### Describe the feature you'd like

I have a project that uses an esproj file instead of csproj, but these files apparently are not supported by csdevkit.

Are there any plans to add this functionality in the future?

**Comment by timheuer** [comment]
Hi @jpringle1 not in the near term.  Given esproj represents a JavaScript project, the environment is generally already supported as just a workspace in VSCode.  I'd love to better understand if it is just rendering in Solution Explorer view here that you'd like or if you've made esproj specific customizations that don't work in VSCode to your liking (versus tasks/launch.json files)

**Comment by jpringle1** [comment]
> Hi @jpringle1 not in the near term. Given esproj represents a JavaScript project, the environment is generally already supported as just a workspace in VSCode. I'd love to better understand if it is just rendering in Solution Explorer view here that you'd like or if you've made esproj specific customizations that don't work in VSCode to your liking (versus tasks/launch.json files)

It's just rendering in the Solution Explorer i'm looking for, to keep a consistent workflow. It's a shame I won't get that anytime soon but fair enough

**Comment by timheuer** [comment]
> It's a shame I won't get that anytime soon but fair enough

It's a fair suggestion -- we'll keep it open/active.  I'm on that same team for esproj as well and can make sure we discuss.

**Comment by bard83** [comment]
I'd also appreciate this feature, even though I've a concern:  what's the meaning of having a solution file which is treated as garbage container for VS? `.esproj` is neither supported by `dotnet sln add ...`

In my case would be nice if the esproj could be at least navigated in the solution explorer view (maybe neglecting the dependencies). In addition would be nice to have the possibility to add to the solution file such proj files. Actually I had to modify manually my solution file based on this [hack](https://stackoverflow.com/questions/59500961/how-to-add-dcproj-into-solution-via-dotnet-sdk).

**Comment by agdgb** [comment]
It's not a good idea to keep a valid project file unsupported in my view. Not sure if this is another way of pushing us to visual studio.

**Comment by agdgb** [comment]
> I'd also appreciate this feature, even though I've a concern: what's the meaning of having a solution file which is treated as garbage container for VS? `.esproj` is neither supported by `dotnet sln add ...`
> 
> In my case would be nice if the esproj could be at least navigated in the solution explorer view (maybe neglecting the dependencies). In addition would be nice to have the possibility to add to the solution file such proj files. Actually I had to modify manually my solution file based on this [hack](https://stackoverflow.com/questions/59500961/how-to-add-dcproj-into-solution-via-dotnet-sdk).

Plus 1 for "IT SHOULD BE NAVIGATABLE IN THE Solution Explorer".

**Comment by kdcllc** [comment]
@timheuer I am not sure why this VS template is specific to VS.NET.  It would be great to have this feature to be part of VSCode for sure, to have consistency.

**Comment by solrevdev** [comment]
> I'm not sure why this VS template is specific to Visual Studio .NET. It would be great to have this feature in VS Code for consistency.

Agreed. Also, the latest SPA templates seem to be only for Windows/Visual Studio users.

See the following links for more information:

- https://learn.microsoft.com/en-us/aspnet/core/client-side/spa/intro?view=aspnetcore-8.0#aspnet-core-spa-templates
- https://learn.microsoft.com/en-us/visualstudio/javascript/tutorial-asp-net-core-with-vue?view=vs-2022

It would be nice if developers (especially front-end) who use VS Code to build React, VueJS, and Svelte apps could have the same or a similar template for using .NET as the API backend, as seems to be available for Visual Studio/Windows.

---

Just to clarify, my intention here is to provide constructive feedback and a suggestion that I believe could benefit a wider range of developers. I hope this comes across as a positive contribution rather than a complaint!

**Comment by abiezerm** [comment]
is this in any plan ?

**Comment by timheuer** [comment]
Not currently in any near term plan.

**Comment by jan-hlavsa-mindplexic** [comment]
It is quite a problem for real world usage. Please consider to add it to the roadmap.
